### PR TITLE
chore(ci): add ffi bindings to CI

### PR DESCRIPTION
**Pull request by Arqu**
## Description

Getting started on adding ffi bindings to CI.
Ref https://github.com/n0-computer/iroh/issues/1264

## Notes & open questions

What platforms should we cover, do we just target macOS for now?
Is this even correct?

## Change checklist

- [x] Self-review.
- [ ] Documentation updates if relevant.
- [ ] Tests if relevant.

### CODE REVIEW

File: .github/workflows/bindings.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+  push:
+    branches:
+      - main
+
+env:
+  RUST_BACKTRACE: 1
+  RUSTFLAGS: -Dwarnings
+  RUSTDOCFLAGS: -Dwarnings
+  MSRV: "1.70"
```
Comment (dignifiedquire) [inline .github/workflows/bindings.yml:13]: current MSRV is `1.66`

Comment (Arqu) [reply .github/workflows/bindings.yml:13]: Ahh, nice, missed that PR.

File: .github/workflows/bindings.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  MSRV: "1.66"
+
+jobs:
+  build_and_test_ffi_bindings:
+    name: Build and test ffi bindings
+    runs-on: ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [macOS-latest]
```
Comment (Arqu) [inline .github/workflows/bindings.yml:22]: should we cover more platforms or is macOS just fine for now?

File: .github/workflows/bindings.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+      - name: Install ${{ matrix.rust }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.rust }}
+
+      - name: Create a copy of iroh.h
+        run: |
+          cd iroh-ffi
+          cp iroh.h iroh.compare.h
```
Comment (dignifiedquire) [inline .github/workflows/bindings.yml:38]: alternatively you can use git diff

Comment (Arqu) [reply .github/workflows/bindings.yml:38]: Eh, since `iroh.h` is in `.gitignore` it doesn't seem to diff it. Also think cmp is slightly faster.

**Pull request merged**